### PR TITLE
Data handling

### LINKED ISSUE #34: Data from Database and Api
# As a dev, 
I want to get more data from my own database, 
so that I'm not that dependent on an external API.

# Description
When I get data I check if it's available in my own Database and if not only then fetch it.
I also want to Post data on the Database the first time I fetch it from the API.

# Done
- [x] Rewrite Routes in backend
- [x] Check if data is already in Database
- [x] Create new user routes
- [x] Create route for Watch and Completed lists

### PR DESCRIPTION
**Pull request by mschmauder-hub**
Fix #34 
I updated the way I format the data I get from my external API, so that I can avoid a few problems when I'm working with my database. There is a limit of request for the external API, that is why I copy the data from the API and post to my Database the first time I fetch it. I intend to create a function to update the data on the database regularly, but I will do this at a later point.

### CODE REVIEW

File: SeriesMaster-Backend/lib/showsDB.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+async function findShows(database, showsId) {
+  const collection = database.collection("shows");
+  const shows = await collection.find({ id: { $in: showsId } }).toArray();
+
+  return shows;
+}
+
+exports.postShow = postShow;
+exports.findShow = findShow;
+exports.findShows = findShows;
```
Comment (leonp5) [inline SeriesMaster-Backend/lib/showsDB.js]: Alternatively you can export it like this:

module.exports = {
postShow,
findShow,
findShows
}

**Review by leonp5 — APPROVED** [review]

File: SeriesMaster-Backend/routes/shows.js [hunk 2]
```diff
@@ -18,7 +19,13 @@ function createShowsRouter() {
   router.get("/:id", async (req, res) => {
     try {
       const id = req.params.id;
-      const show = await searchSingleShow(id);
+      let show;
+      show = await findShow(database, id);
+
+      if (!show) {
+        show = await searchSingleShow(id);
```
Comment (leonp5) [inline SeriesMaster-Backend/routes/shows.js:26]: I'm not sure if i understand the code right, but with this id, you passing through `searchSingleShow` you get the intended show from the API?

File: SeriesMaster-Backend/lib/showsDB.js (hunk 1 above)
Comment (mschmauder-hub) [reply SeriesMaster-Backend/lib/showsDB.js]: Thank you!

File: SeriesMaster-Backend/routes/shows.js (hunk 2 above)
Comment (mschmauder-hub) [reply SeriesMaster-Backend/routes/shows.js:26]: Yes, that is correct. Maybe I need to name it more clearly.

**Pull request merged**